std/algorithm/iteration.d

```diff
@@ -955,11 +920,19 @@ private:
          isForeachUnaryWithIndexIterable!R);
 
 public:
+    Flag!"each" each(Range)(auto ref Range r)
+    if (isForeachIterable!Range ||
```

Comment:
Largely unsuccessful attempt to make the constraint nicer. Will get back to this.